dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -84,28 +84,26 @@ def read_metadata(
         filters=None,
         **kwargs
     ):
+        parts = []
         if len(paths) > 1:
             if gather_statistics is not False:
                 # this scans all the files, allowing index/divisions
                 # and filtering
-                pf = fastparquet.ParquetFile(
+                pf = ParquetFile(
                     paths, open_with=fs.open, sep=fs.sep, **kwargs.get("file", {})
                 )
             else:
-                base, fns = analyse_paths(paths)
+                # Rely on metadata for 0th file.
```

Comment:
This branch could also be useful in the case that a directory was passed, whether it contains only data files, or happens to have a `_common_metadata` file (i.e., one with just the schema and no row-group information)

Reply:
I revised `read_metadata` a bit to account for the cases you suggested.  I also added a new test for passing a directory to `read_parquet` without a `_metadata` file in the directory.   Feel free to poke holes :)